neuralop/data/transforms/rescale.py

```diff
@@ -0,0 +1,372 @@
+import torch
+import numpy as np
+import torchvision.transforms.functional as F
+
+
+def DarcyExtractBC(x, y):
+    """
+    Extract boundary conditions for Darcy Equation: 2D boundary value problem.
+    Code adapted from paper https://arxiv.org/abs/2507.18813
+
+    Attributes
+    ----------
+    x: torch.Tensor
+        initial condition, a tensor of shape (batch, channels, X, Y)
+    y: torch.Tensor
+        boundary condition, a tensor of shape (batch, channels, X, Y)
+
+    Returns
+    -------
+    x: torch.Tensor
+        the combined input tensor with boundary conditions added in channels 1,...,4
+    """
+    unsqueeze_x = x.ndim == 3  # check whether batch dimension is missing
+    unsqueeze_y = y.ndim == 3
+
+    # add batch dimension if there was no batch dimension
+    if unsqueeze_x:
+        x = x.unsqueeze(0)
+    if unsqueeze_y:
+        y = y.unsqueeze(0)
+
+    # extract BC
+    Nx, Ny = y.shape[-2], y.shape[-1]
```

Comment:
Some documentation of what the BC is/what this extracts might be helpful. Perhaps one line